backend/backends/datadog/datadog.go

```diff
@@ -216,12 +216,13 @@ func (d *client) authenticatedURL(path string) string {
 
 // NewClientFromViper returns a new Datadog API client.
 func NewClientFromViper(v *viper.Viper) (backendTypes.Backend, error) {
-	v.SetDefault("datadog.timeout", defaultClientTimeout)
-	v.SetDefault("datadog.max_request_elapsed_time", defaultMaxRequestElapsedTime)
+	dd := getSubViper(v, "datadog")
+	dd.SetDefault("timeout", defaultClientTimeout)
+	dd.SetDefault("max_request_elapsed_time", defaultMaxRequestElapsedTime)
 	return NewClient(
-		v.GetString("datadog.api_key"),
-		v.GetDuration("datadog.timeout"),
-		v.GetDuration("datadog.max_request_elapsed_time"),
+		dd.GetString("api_key"),
+		dd.GetDuration("timeout"),
+		dd.GetDuration("max_request_elapsed_time"),
```

Comment:
There are many bugreports about this/similar issue:
https://github.com/spf13/viper/pull/165
https://github.com/spf13/viper/issues/71
https://github.com/spf13/viper/issues/158
https://github.com/spf13/viper/issues/160
https://github.com/spf13/viper/issues/168
https://github.com/spf13/viper/issues/191

Sadly looks like nobody cares.